### PR TITLE
Web integration

### LINKED ISSUE #173: Merge core-web
Merge core-web into this library, including its build process.

### PR DESCRIPTION
**Pull request by perry-mitchell**
Fixes #173

### CODE REVIEW

File: source/web/HashingTools.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+"use strict";
+
+function arrayBufferToHexString(arrayBuffer) {
+    var byteArray = new Uint8Array(arrayBuffer);
```
Comment (sallar) [inline source/web/HashingTools.js]: Would be nice to have these as const and let now that we are updating?

File: source/web/HashingTools.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+"use strict";
```
Comment (sallar) [inline source/web/HashingTools.js]: Do we need use strict? I believe this file is a bit old

File: source/web/HashingTools.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+ * Perform patching of the PBKDF2 function in iocane
+ * @param {Function|undefined=} handler Optionally override the internal PBKDF2 engine
+ */
+function patchCorePBKDF(handler = lib.deriveKeyFromPassword) {
+    const { vendor } = require("./index.js");
+    vendor.iocane.components.setPBKDF2(handler);
+}
+
+function joinBuffers(buffer1, buffer2) {
+    var tmp = new Uint8Array(buffer1.byteLength + buffer2.byteLength);
```
Comment (sallar) [inline source/web/HashingTools.js]: Also here and the next function

File: test/index.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ var chai = require("chai"),
     sinon = require("sinon"),
     mocha = require("mocha");
 
-var Buttercup = require("../source/module.js"),
+var Buttercup = require("../source/node/index.js"),
```
Comment (sallar) [inline test/index.js]: should we also `const` here?

File: tests/ArchiveComparatorTests.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
-var lib = require("../source/module.js");
+var lib = require("../source/node/index.js");
```
Comment (sallar) [inline tests/ArchiveComparatorTests.js]: Some of the tests also use `var`. its not important but is it easy to change those as well?

File: source/web/HashingTools.js (hunk 2 above)
Comment (perry-mitchell) [reply source/web/HashingTools.js]: Nope! We do not. I'll remove it.

File: source/web/HashingTools.js (hunk 1 above)
Comment (perry-mitchell) [reply source/web/HashingTools.js]: Yes! Agreed. Case of the copy+paste.. haha. I'll change it.

File: tests/ArchiveComparatorTests.js (hunk 5 above)
Comment (perry-mitchell) [reply tests/ArchiveComparatorTests.js]: Maybe these could be part of a later refactor?

**Pull request merged**